### ISSUE TITLE
[proposal] unzip function

### ISSUE THREAD

**Issue by timonson** [selfrefactor/rambda#317]
I love this library and I would like to propose an `unzip` function. It does pretty much the opposite of what your `zip` function does. I use this one:

```javascript
function unzip(array) {
  return array.reduce((acc, el) => {
    el.forEach((nestedEl, i) =>
      Array.isArray(acc[i]) ? acc[i].push(nestedEl) : acc.push([nestedEl])
    )
    return acc
  }, [])
}

const array = [[10, "A"], [20, "B"], [30, "C", 999], [40, "D"]]
unzip(array)  //  [ [ 10, 20, 30, 40 ], [ "A", "B", "C", "D" ], [ 999 ] ]
```

What do you think? By the way, ramda has a similar function called `transpose`.

**Comment by selfrefactor** [comment]
Thank you for the approving words. I will test it out and get back to you.

**Comment by selfrefactor** [comment]
As this method is not part of `Ramda`, the only way I can add it is to `Rambdax` which is the extended version of `Rambda`. If that is fine with you, please let me know.

**Comment by timonson** [comment]
But ramda has this function, it is called `transpose` there:

```javascript
const array = [[10, "A"], [20, "B"], [30, "C", 999], [40, "D"]]
console.log(R.transpose(array)) //  [ [ 10, 20, 30, 40 ], [ "A", "B", "C", "D" ], [ 999 ] ]
```

**Comment by selfrefactor** [comment]
Now I get it. What I will do is run your implementation through the `Ramda` spec and will get back to you with the outcome. If it passes all tests, then it is clear what happens next.

**Comment by selfrefactor** [comment]
The test are passing, so the method `R.transpose` will be added in next `4.3.0` release. Naming it `unzip` might sound tempting, but I'd prefer not to increase differences between Rambda and Ramda.

**Comment by timonson** [comment]
That's amazing, thank you! I like both names :)

**Comment by selfrefactor** [comment]
Published with `4.4.0`
